dask/utils.py

```diff
@@ -166,75 +186,81 @@ def get_bin_linesep(encoding, linesep):
         return linesep.encode(encoding)
 
 
-def next_linesep(fo, seek, encoding, linesep):
-    bin_linesep = get_bin_linesep(encoding, linesep)
-    data = b''
-    data_len = 0
-
-    while bin_linesep not in data:
-        data_len += 1
-        fo.seek(seek)
-        data = fo.read(data_len)
-        if len(data) < data_len:
-            break  # eof
-
-    stop = seek + data_len
-    start = stop - len(bin_linesep)
-    return start, stop
-
-
-def textblock(file, start, stop, compression=None, encoding=None,
-              linesep=None):
-    """ Pull out a block of text from a file given start and stop bytes
+def textblock(filename, start, end, compression=None, encoding=system_encoding,
+              linesep=os.linesep):
+    """Pull out a block of text from a file given start and stop bytes.
 
-    This gets data starting/ending from the next linesep delimiter
+    This gets data starting/ending from the next linesep delimiter.
 
     Examples
     --------
-
-    >> with open('myfile.txt', 'w') as f:
+    >> with open('myfile.txt', 'wb') as f:
     ..     f.write('123\n456\n789\nabc')
 
-    >> f = open('myfile.txt')
-
-    In the example below, 1 and 10 don't line up with endlines
+    In the example below, 1 and 10 don't line up with endlines.
 
-    >> textblock(f, 1, 10)
+    >> u''.join(textblock('myfile.txt', 1, 10))
     '456\n789\n'
     """
-    if isinstance(file, (str, unicode)):
-        myopen = opens.get(compression, open)
-        f = myopen(file, 'rb')
-        try:
-            result = textblock(f, start, stop, compression=None,
-                               encoding=encoding, linesep=linesep)
-        finally:
-            f.close()
-        return result
+    # Make sure `linesep` is not a byte string because `io.TextIOWrapper` in
+    # Python versions other than 2.7 dislike byte strings for the `newline`
+    # argument.
+    linesep = str(linesep)
 
-    if encoding is None:
-        encoding = getattr(file, 'encoding', 'utf-8')
-        if encoding is None:
-            encoding = 'utf-8'
-
-    if linesep is None:
-        linesep = os.linesep
-
-    if start:
-        startstart, startstop = next_linesep(file, start, encoding, linesep)
-    else:
-        startstart = start
-        startstop = start
-
-    if stop is None:
-        file.seek(start)
-        return file.read()
-
-    stopstart, stopstop = next_linesep(file, stop, encoding, linesep)
-
-    file.seek(startstop)
-
-    return file.read(stopstop - startstop)
+    chunksize = end - start + 1
```

Comment:
What is the correct chunk size? If ``start`` and ``end`` had the same semantics as in ``slice``, then the chunk size would be ``end - start``. If ``start`` and ``end`` were both inclusive (which does not make sense for splitting up a larger file into chunks), it would be ``end - start + 1``, hence, I think the current chunk size is actually wrong. From the docstring and the implementation, the first line that ``textblock`` yields is the one *after* the next line separator. Thus, even if ``start`` happened to point to the first character of a line, that line would not be part of the block (except ``start`` is 0). On the other hand, if ``start`` happened to point to the (first character of the) line separator of a previous line and ``end`` happened to point to the first character of some later line, the chunk size would be ``end - start - len(linesep)`` because we definitely need to advance the ``start`` pointer by ``len(linesep)``. So, is the (maximum) chunk size given ``start`` and ``end`` in fact ``end - start - len(linesep)``? All of this is mostly only important for the exception in line 258, but I feel we should get this right before merging the PR. Could you briefly comment on my thoughts to make sure they make sense?